src/BuiltInTools/dotnet-watch/HotReload/BlazorWebAssemblyHostedDeltaApplier.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.ExternalAccess.Watch.Api;
+using Microsoft.Extensions.Tools.Internal;
+
+namespace Microsoft.DotNet.Watcher.Tools
+{
+    internal class BlazorWebAssemblyHostedDeltaApplier : IDeltaApplier
+    {
+        private readonly BlazorWebAssemblyDeltaApplier _wasmApplier;
+        private readonly AspNetCoreDeltaApplier _hostApplier;
+
+        public BlazorWebAssemblyHostedDeltaApplier(IReporter reporter)
+        {
+            _wasmApplier = new BlazorWebAssemblyDeltaApplier(reporter);
+            _hostApplier = new AspNetCoreDeltaApplier(reporter)
+            {
+                SuppressBrowserRefreshAfterApply = true,
```

Comment:
This flag is saying that if an update was made to the host then browser doesn't need to refresh because unlike other AspNetCore scenarios the app being served in the browser is not running in the host but in the Blazor WASM side.

Is my understanding of that correct?

Reply:
By default `AspNetCoreDeltaApplier` will attempt to either re-render components or refresh the browser after a delta has been applied. Since the WebAssembly one takes care of re-rendering, we kinda want this to no-op.